Cubical/M-types/helper.agda

```diff
@@ -0,0 +1,317 @@
+{-# OPTIONS --cubical --guardedness #-} --safe
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv using (_≃_)
+open import Cubical.Foundations.Function using (_∘_)
+
+open import Cubical.Data.Unit
+open import Cubical.Data.Prod
+open import Cubical.Data.Nat as ℕ using (ℕ ; suc ; _+_ )
+open import Cubical.Data.Sigma
+
+open import Cubical.Foundations.Transport
+open import Cubical.Foundations.Univalence
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Function
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.GroupoidLaws
+open import Cubical.Foundations.Path
+open import Cubical.Foundations.Embedding
+open import Cubical.Foundations.FunExtEquiv
+
+module Cubical.M-types.helper where
```

Comment:
This file has lots great lemmas, I would love to see most of them upstreamed. @mortberg what do you think –perhaps the sigma lemmas could be moved to `Sigma.Properties` and the inj lemmas to `Functions.Embedding`?  Things like `diagonal-unit`, `funExtIso` have obvious homes.

Reply:
If will try and clean up the pull request, when I have completed the last postulate. I would also like to figure a way of not having the abstract block, but I am not sure that is possible/easy.

Reply:
Yeah, all general results should definitely be upstreamed